### PR TITLE
Server rendering for widget embed

### PR DESCRIPTION
**Pull request by mluena**
## Overview
MyRW: Cancel button from Edit dashboard form function changed from window.history goBack()  to go(-1)

## Testing instructions
Go to /myrw-detail/dashboards/ , try to clone one and press cancel in Edit view. Try with ther cancel btn

## Pivotal task
https://www.pivotaltracker.com/story/show/164955692

---

## Checklist before submitting
- [ ] Title: Don't forget to make it clear for everyone, even for people that don't know about the feature/bug.
- [ ] Meaningful commits and code rebased on `develop`.

### CODE REVIEW

File: components/form/Navigation.js [hunk 1]
```diff
@@ -6,6 +6,19 @@ import Button from 'components/ui/Button';
...
+    stepLength: PropTypes.number.isRequired,
+    submitting: PropTypes.bool.isRequired,
+    hideCancel: PropTypes.bool,
+    onStepChange: PropTypes.func.isRequired,
+    cancel: PropTypes.func
+  }
+
+  static defaultProps = {
+    hideCancel: false,
+    cancel: null
```
Comment (andresgnlez) [inline components/form/Navigation.js]: https://github.com/resource-watch/resource-watch/pull/936/files/1613d8556dcbe0871eff52c175d9e4bf4557a3bf..f6f6a2bed9d6194544153d73aab8e5aa8bbae89f#r271004582

File: pages/app/embed/map/index.js [hunk 2]
```diff
@@ -14,17 +14,18 @@ import { setEmbed, setWebshotMode } from 'redactions/common';
 import EmbedMapPage from './component';
 
 class EmbedMapPageContainer extends PureComponent {
+
```
Comment (andresgnlez) [inline pages/app/embed/map/index.js]: Remove extra line.

File: redactions/widget.js [hunk 3]
```diff
@@ -186,7 +186,7 @@ function fetchRasterBandInfo(datasetId, bandName) {
  * @param {string} layerId Layer ID
  */
 function fetchLayer(datasetId, layerId) {
```
Comment (andresgnlez) [inline redactions/widget.js]: Change this to `getLayer`, `fetchLayer` should be keeped.

getX => internal call in the app.
fetchX => external call to other services.

File: services/LayersService.js [hunk 4]
```diff
@@ -175,3 +150,45 @@ export default class LayersService {
     });
   }
 }
+
+/**
+ * Fetches a layer according to widget id and params.
+ *
+ * @param {String} id - widget id.
```
Comment (andresgnlez) [inline services/LayersService.js]: Update docs.

File: layout/app/pulse/layer-pill/actions.js [hunk 5]
```diff
@@ -24,8 +24,7 @@ export const toggleContextualLayer = createThunkAction('layer-pill/toggleContext
 
     if (!layerFound) {
       dispatch(setContextLayersLoading(true));
-      const layersService = new LayersService();
-      await layersService.fetchData({ id })
+      await dispatch(fetchLayer({ id }))
```
Comment (andresgnlez) [inline layout/app/pulse/layer-pill/actions.js]: Isn't the function paramater supposed to be `(id, params)` ? Being `id` an string, why do you send an object with a key `id` ?

File: components/widgets/list/WidgetCard.js [hunk 6]
```diff
@@ -41,7 +41,7 @@ import LoginRequired from 'components/ui/login-required';
 // Services
 import WidgetService from 'services/WidgetService';
 import UserService from 'services/UserService';
-import LayersService from 'services/LayersService';
+import { fetchLayer as fetchData } from 'services/LayersService';
```
Comment (andresgnlez) [inline components/widgets/list/WidgetCard.js]: Rename.

File: components/admin/data/layers/form/LayersForm.js [hunk 7]
```diff
@@ -73,7 +67,7 @@ class LayersForm extends React.Component {
 
     // Add the dashboard promise if the id exists
     if (id) {
-      promises.push(this.service.fetchData({ id }));
+      promises.push(fetchLayer({ id }));
```
Comment (andresgnlez) [inline components/admin/data/layers/form/LayersForm.js]: This shouldn't work according to your param input.

File: services/LayersService.js (hunk 4 above)
Comment (andresgnlez) [inline services/LayersService.js]: Update docs.

File: services/LayersService.js [hunk 8]
```diff
@@ -175,3 +150,45 @@ export default class LayersService {
...
+    transformResponse: [].concat(
+      WRIAPI.defaults.transformResponse,
+      ({ data }) => data
+    )
+  }).then((response) => {
+    const { status, statusText, data } = response;
+    logger.debug(`Widget layer fetch returned with code ${status}`);
+
+    if (status >= 300) {
+      logger.error('Error fetching layer:', `${status}: ${statusText}`);
```
Comment (andresgnlez) [inline services/LayersService.js]: Add `404` error exception to be a `debug`, not an error.

File: services/LayersService.js [hunk 9]
```diff
@@ -175,3 +150,47 @@ export default class LayersService {
...
+      application: process.env.APPLICATIONS,
+      ...params
+    },
+    transformResponse: [].concat(
+      WRIAPI.defaults.transformResponse,
+      ({ data }) => data
+    )
+  }).then((response) => {
+    const { status, statusText, data } = response;
+    logger.debug(`Widget layer fetch returned with code ${status}`);
```
Comment (andresgnlez) [inline services/LayersService.js]: Do not debug this, only possible errors of beginning of actions, remember me tomorrow to tell you more about this.

**Comment by mluena** [comment]
Ok! 🍺

**Pull request merged**